pdns/dbdnsseckeeper.cc

```diff
@@ -558,6 +559,181 @@ bool DNSSECKeeper::getTSIGForAccess(const DNSName& zone, const string& master, D
   return false;
 }
 
+bool DNSSECKeeper::rectifyZone(const DNSName& zone, string& error) {
+  if (isPresigned(zone)) {
+    error =  "Rectify presigned zone '"+zone.toLogString()+"' is not allowed/necessary.";
+    return false;
+  }
+
+  UeberBackend B("default");
```

Comment:
DNSSECKeeper is supposed to have an UeberBackend instance already, so can you reuse that?

Reply:
Will do!

Reply:
Interestingly, re-using the UeberBackend in the DNSSECKeeper makes the bind-dnssec-both test fail because `getSOAUncached()` returns false for secure-delegated.dnssec-parent.com. Investigating

Reply:
confirmed, reverting 1d7630f631b1e81373d158021a9eb1daa9442762 makes all the tests pass.

Reply:
That sounds reaaaally bad. Like something keeping a list() operation open or sth.

Reply:
ah, I think I got it. In pdnsutil.cc:1983 we create a DNSSECKeeper object _without_ handing it an UeberBackend. So when rectifyZone (in pdnsutil) is called we indeed can not get a SOA from that dk.

Reply:
so this is a bug(ish) in pdnsutil, but it does mean that calling `DNSSECKeeper::rectifyZone` means we need the Ueber backend. Maybe it indeed makes more sense to move rectifyZone to the UeberBackend (but that has other pain points)

Reply:
Alright, I have made a solution that uses the existing UeberBackend if it works and allocates a new one when needed. C++ wizards are welcome to make this better......